### PR TITLE
Lint GitHub Actions and add all linters explicitly to CI

### PR DESCRIPTION
**Pull request by tdeebswihart**
## What changed?
I added the [actionlint](https://github.com/rhysd/actionlint) GHA linter, unified how we call linters from our makefile, and added explicit (and parallel) GHA jobs for each linter.

## Why?

To prevent us from committing GHA bugs

## How did you test it?
CI and locally using ACT (in progress!)

## Potential risks
None

## Documentation
No

## Is hotfix candidate?
No

### CODE REVIEW

**Comment by tdeebswihart** [comment]
I'd _like_ to move the golangci linter to `lint-code` but I'm no longer able to configure our required actions, so I left it alone

File: .github/actionlint-matcher.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "problemMatcher": [
+    {
+      "owner": "actionlint",
+      "pattern": [
+        {
+          "regexp": "^(?:\\x1b\\[\\d+m)?(.+?)(?:\\x1b\\[\\d+m)*:(?:\\x1b\\[\\d+m)*(\\d+)(?:\\x1b\\[\\d+m)*:(?:\\x1b\\[\\d+m)*(\\d+)(?:\\x1b\\[\\d+m)*: (?:\\x1b\\[\\d+m)*(.+?)(?:\\x1b\\[\\d+m)* \\[(.+?)\\]$",
```
Comment (stephanos) [inline .github/actionlint-matcher.json:7]: did you write that? 😅

**Review by stephanos — APPROVED** [review]
:chefskiss:

File: .github/actionlint-matcher.json (hunk 1 above)
Comment (tdeebswihart) [reply .github/actionlint-matcher.json:7]: Oh heaven's no, that came from the actionlint repo itself 😛

File: Makefile [hunk 2]
```diff
@@ -318,16 +324,23 @@ goimports:
...
-lint: $(GOLANGCI_LINT)
+lint-actions: $(ACTIONLINT)
+	@printf $(COLOR) "Linting GitHub actions..."
+	@$(ACTIONLINT)
+
+lint-code: $(GOLANGCI_LINT)
+	@printf $(COLOR) "Linting code..."
+	@$(GOLANGCI_LINT) run --verbose --timeout 10m --fix=$(GOLANGCI_LINT_FIX) --new-from-rev=$(GOLANGCI_LINT_BASE_REV) --config=.golangci.yml
+
+lint: lint-code lint-actions lint-api lint-protos
```
Comment (rodrigozhou) [inline Makefile:335]: Won't this run the deps first, and then the print below? Also, why not call `lint-api` and `lint-protos` here too?

File: .github/actionlint-matcher.json (hunk 1 above)
Comment (rodrigozhou) [reply .github/actionlint-matcher.json:7]: What is this?

**Pull request merged**